### PR TITLE
Rename connection to tunnel

### PR DESCRIPTION
**Pull request by alalamav**
- Renames connection to tunnel in `cordova.plugin.outline` and native platform implementations.
- Minor documentation updates.
- I have left TODOs to rename storage keys when we perform a data migration (i.e. update server model schema), in order to make this change only a code refactor.

### CODE REVIEW

File: src/electron/index.ts [hunk 1]
```diff
@@ -298,26 +300,26 @@ async function startVpn(
     throw new Error('already connected');
   }
 
-  currentConnection = new ConnectionManager(config, isAutoConnect);
+  currentConnection = new TunnelManager(config, isAutoConnect);
```
Comment (JonathanDCohen) [inline src/electron/index.ts:303]: currentTunnel?

File: cordova-plugin-outline/README.md [hunk 2]
```diff
@@ -44,17 +44,17 @@ We use a custom framework dependency, PacketProcessor, that implements tun2socks
...
-This plugin targets Apple devices running iOS 9.0+ and macOS/OS X 10.11.
+This plugin targets Apple devices running iOS 11.0+ and macOS/OS X 10.11.
 
 ## JavaScript API
 
 ```ts
 
 // Represents a Shadowsocks server configuration.
-interface ShadowsocksServerConfig {
+interface ServerConfig {
```
Comment (JonathanDCohen) [inline cordova-plugin-outline/README.md:57]: Did you mean to change this name as well?

**Review by JonathanDCohen — COMMENTED** [review]
Can you leave some thoughts as to why Tunnel makes more sense than Connection?  Because we can multiplex connections over a single tunnel?

File: cordova-plugin-outline/README.md (hunk 2 above)
Comment (alalamav) [reply cordova-plugin-outline/README.md:57]: Yes, this interface was always `ServerConfig`. The documentation was not in sync.

File: src/electron/index.ts (hunk 1 above)
Comment (alalamav) [reply src/electron/index.ts]: Good catch, done in 07fcf56.

**Comment by alalamav** [comment]
> Can you leave some thoughts as to why Tunnel makes more sense than Connection? Because we can multiplex connections over a single tunnel?

That's right. I think the abstraction fits better with networking terminology (i.e. "apps establish TCP connections over a VPN tunnel", TUN device) and native platform concepts (i.e. Apple's [PacketTunnelProvider](https://developer.apple.com/documentation/networkextension/nepackettunnelprovider), our own Android VpnTunnelService). 

Thank you for the review :)

**Review by JonathanDCohen — APPROVED** [review]

**Pull request merged**